lib/historical_link_checker.rb

```diff
@@ -0,0 +1,77 @@
+class HistoricalLinkChecker
+  attr_reader :report
+
+  def initialize(uri, checks)
+    @uri = uri
+    @checks = sorted_checks(checks)
+    @report = LinkChecker::UriChecker::Report.new
+  end
+
+  def call
+    current_report = LinkChecker.new(uri).call
+
+    if needs_historical_data_check?(current_report)
+      check_history(current_report)
+    else
+      current_report
+    end
+  end
+
+private
+
+  attr_reader :uri, :checks
+
+  def check_history(current_report)
```

Comment:
Not sure if missing something, but is there anything that checks whether it is the same problem that is occurring?

As far as I can tell you could have something return 500 for a week and then next return 404 and this would say 404 not found has been occurring for 7 days